### PR TITLE
EventLog Scopes

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/aspnet/Logging/issues/713

### CODE REVIEW

File: test/Microsoft.Extensions.Logging.Test/EventLogLoggerTest.cs [hunk 1]
```diff
@@ -120,6 +120,32 @@ public void MessageWithinMaxSize_WritesFullMessage(int messageSize)
...
         }
 
+
+        [Fact]
+        public void Message_WritesFullMessageWithScopes()
+        {
+            // Arrange
+            var loggerName = "Test";
+            var maxMessageSize = 50 + loggerName.Length + Environment.NewLine.Length;
+            var expectedMessage = loggerName + Environment.NewLine +
```
Comment (BrennanConroy) [inline test/Microsoft.Extensions.Logging.Test/EventLogLoggerTest.cs]: This is how we want messages to look?
```
Test
Message
Outer Scope
Inner Scope
```

File: src/Microsoft.Extensions.Logging.EventLog/EventLogLoggerProvider.cs [hunk 2]
```diff
@@ -9,10 +9,12 @@ namespace Microsoft.Extensions.Logging.EventLog
     /// The provider for the <see cref="EventLogLogger"/>.
     /// </summary>
     [ProviderAlias("EventLog")]
-    public class EventLogLoggerProvider : ILoggerProvider
+    public class EventLogLoggerProvider : ILoggerProvider, ISupportExternalScope
```
[the remark this answers is not in the document]
Comment (pakrym) [reply src/Microsoft.Extensions.Logging.EventLog/EventLogLoggerProvider.cs]: While this is technically a breaking and adding interface can change overload resolution `ISupportExternalScope` would be new in the release so no one would be using it.

[the remark this answers is not in the document]
Comment (BrennanConroy) [reply src/Microsoft.Extensions.Logging.EventLog/EventLogLoggerProvider.cs]: @Eilon

File: test/Microsoft.Extensions.Logging.Test/EventLogLoggerTest.cs (hunk 1 above)
Comment (BrennanConroy) [reply test/Microsoft.Extensions.Logging.Test/EventLogLoggerTest.cs]: @glennc For formatting

File: src/Microsoft.Extensions.Logging.EventLog/EventLogLoggerProvider.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Eilon) [reply src/Microsoft.Extensions.Logging.EventLog/EventLogLoggerProvider.cs]: Yeah not worried about this type of breaking change in this particular case.

[the remark this answers is not in the document]
Comment (Eilon) [reply src/Microsoft.Extensions.Logging.EventLog/EventLogLoggerProvider.cs]: But good to ask!

**Review by muratg — APPROVED** [review]
:shipit:

**Comment by snakefoot** [comment]
I'm curious about this change. When the logging provider (EventLogLoggerProvider) implements `ISupportExternalScope` then it means that the LoggerFactory-implementation will no longer call EventLogLogger.BeginScope. Instead the `EventLogLogger` must use the provided `IExternalScopeProvider` to get its scope-state.

But the EventLogLoggerProvider.GetLogger() doesn't pass the IExternalScopeProvider as parameter to the newly created EventLogLogger. Instead the EventLogLogger creates its own internal version, that is only exercised when calling BeginScope directly, but in the "real world" that will never happen.

Is it the intention that EventLogLogger should not perform logging of current scope?

**Comment by pakrym** [comment]
Good catch. That was a bug that was fixed in the new versions: https://github.com/aspnet/Extensions/blob/master/src/Logging/Logging.EventLog/src/EventLogLoggerProvider.cs#L36

**Comment by snakefoot** [comment]
Ahh it was hidden in the refactoring commit from https://github.com/aspnet/Extensions/pull/502. Sorry about the noise

**Comment by snakefoot** [comment]
Any reasons for having the stale copy around here? when the correct version now resides at new location?

**Comment by pakrym** [comment]
Repository title says `[Archived] Common logging abstractions and a few implementations. Project moved to https://github.com/aspnet/Extensions`

**Pull request merged**